### PR TITLE
fix(geometry): keep the W axis in unproject_meshgrid when W is 1

### LINKED ISSUE #4278: W1b: `unproject_meshgrid` — `.squeeze()` destroys the `W = 1` axis, corrupting `depth_to_3d_v2` and `warp_frame_depth`
`unproject_meshgrid` calls `.squeeze()` on the meshgrid it builds. With `W = 1` that removes the width axis, and the grid then broadcasts across a phantom width — so `unproject_meshgrid(1, 3, K)` and `unproject_meshgrid(3, 1, K)` return the **same shape** `(1, 1, 3, 3)` with different values, violating the documented `(*, H, W, 3)` contract. `H = 1` is unaffected (the batch axis absorbs the squeeze), and `H = W = 1` is correct by accident.

Three public functions inherit it:

- **`depth_to_3d_v2` with `W = 1`** returns `(1, 3, 3, 3)` where `depth_to_3d` returns the correct `(1, 3, 1, 3)`; the values are the correct column tiled three times. This is the one input at which the two implementations, otherwise `torch.equal` under asymmetric `K`, `H ≠ W`, batching and float64, disagree.
- **`warp_frame_depth` with `W = 1`** returns a `(1, 2, 4, 4)` image for a `(1, 2, 4, 1)` input — a silently wrong-width result. `H = 1` is fine.
- **`depth_to_normals` with `W = 1`** returns a `(1, 3, 4, 4)` normal map for a `(1, 1, 4, 1)` depth, with its single column of normals laid along the phantom width. `H = 1` and `H = W = 1` are fine.

`W = 1` is a degenerate but perfectly valid shape (a single image column, a 1-pixel-wide crop), and nothing raises.

The `.squeeze()` is on the line that #2709 (closed 2025-05-07) and #2970 (closed 2025-12-29) touched when `unproject_meshgrid` was made batched; see also #4271 for the docstring half.

## Repro

```python
import torch
from kornia.geometry.depth import depth_to_3d, depth_to_3d_v2, depth_to_normals, unproject_meshgrid, warp_frame_depth

K = torch.tensor([[[100.0, 0.0, 4.0], [0.0, 100.0, 3.0], [0.0, 0.0, 1.0]]])

print("unproject_meshgrid(H=1, W=3):", tuple(unproject_meshgrid(1, 3, K).shape))
print("unproject_meshgrid(H=3, W=1):", tuple(unproject_meshgrid(3, 1, K).shape), " <- same shape, documented (*, H, W, 3)")
print("  H=1,W=3 values:", unproject_meshgrid(1, 3, K).flatten().tolist())
print("  H=3,W=1 values:", unproject_meshgrid(3, 1, K).flatten().tolist())
print("unproject_meshgrid(H=1, W=1):", tuple(unproject_meshgrid(1, 1, K).shape))

d = torch.full((1, 1, 3, 1), 2.0)                    # B=1, H=3, W=1
v1 = depth_to_3d(d, K)
v2 = depth_to_3d_v2(d[:, 0], K)
print("depth_to_3d(W=1)    shape:", tuple(v1.shape), "-> (B, 3, H, W)")
print("depth_to_3d_v2(W=1) shape:", tuple(v2.shape), "-> should be (B, H, W, 3) = (1, 3, 1, 3)")
print("depth_to_3d_v2(W=1) values:", v2.flatten().tolist())
print("depth_to_3d(W=1)    values:", v1[0].permute(1, 2, 0).flatten().tolist())
dh = torch.full((1, 1, 1, 3), 2.0)                   # H=1 is fine
print("H=1: v1.permute == v2  :", torch.equal(depth_to_3d(dh, K)[0].permute(1, 2, 0)[None], depth_to_3d_v2(dh[:, 0], K)))

img = torch.rand(1, 2, 4, 1, generator=torch.Generator().manual_seed(0))
dep = torch.ones(1, 1, 4, 1)
print("warp_frame_depth(W=1) shape:", tuple(warp_frame_depth(img, dep, torch.eye(4)[None], K).shape), "-> should be (1, 2, 4, 1)")
imgh = torch.rand(1, 2, 1, 5, generator=torch.Generator().manual_seed(0))
print("warp_frame_depth(H=1) shape:", tuple(warp_frame_depth(imgh, torch.ones(1, 1, 1, 5), torch.eye(4)[None], K).shape))

col = torch.tensor([[[[1.0], [2.0], [3.0], [4.0]]]])          # B=1, H=4, W=1
nrm = depth_to_normals(col, K)
print("depth_to_normals(W=1) shape:", tuple(nrm.shape), "-> should be (1, 3, 4, 1)")
print("depth_to_normals(W=1) row 0 normals:", [[round(v, 6) for v in nrm[0, :, 0, j].tolist()] for j in range(nrm.shape[-1])])
print("depth_to_normals(H=1) shape:", tuple(depth_to_normals(col.reshape(1, 1, 1, 4), K).shape))
print("depth_to_normals(H=W=1) shape:", tuple(depth_to_normals(col[:, :, :1], K).shape))
```

## Observed

```
unproject_meshgrid(H=1, W=3): (1, 1, 3, 3)
unproject_meshgrid(H=3, W=1): (1, 1, 3, 3)  <- same shape, documented (*, H, W, 3)
  H=1,W=3 values: [-0.03999999910593033, -0.029999999329447746, 1.0, -0.029999999329447746, -0.029999999329447746, 1.0, -0.019999999552965164, -0.029999999329447746, 1.0]
  H=3,W=1 values: [-0.03999999910593033, -0.029999999329447746, 1.0, -0.03999999910593033, -0.019999999552965164, 1.0, -0.03999999910593033, -0.009999999776482582, 1.0]
unproject_meshgrid(H=1, W=1): (1, 1, 1, 3)
depth_to_3d(W=1)    shape: (1, 3, 3, 1) -> (B, 3, H, W)
depth_to_3d_v2(W=1) shape: (1, 3, 3, 3) -> should be (B, H, W, 3) = (1, 3, 1, 3)
depth_to_3d_v2(W=1) values: [-0.07999999821186066, -0.05999999865889549, 2.0, -0.07999999821186066, -0.03999999910593033, 2.0, -0.07999999821186066, -0.019999999552965164, 2.0, -0.07999999821186066, -0.05999999865889549, 2.0, -0.07999999821186066, -0.03999999910593033, 2.0, -0.07999999821186066, -0.019999999552965164, 2.0, -0.07999999821186066, -0.05999999865889549, 2.0, -0.07999999821186066, -0.03999999910593033, 2.0, -0.07999999821186066, -0.019999999552965164, 2.0]
depth_to_3d(W=1)    values: [-0.07999999821186066, -0.05999999865889549, 2.0, -0.07999999821186066, -0.03999999910593033, 2.0, -0.07999999821186066, -0.019999999552965164, 2.0]
H=1: v1.permute == v2  : True
warp_frame_depth(W=1) shape: (1, 2, 4, 4) -> should be (1, 2, 4, 1)
warp_frame_depth(H=1) shape: (1, 2, 1, 5)
depth_to_normals(W=1) shape: (1, 3, 4, 4) -> should be (1, 3, 4, 1)
depth_to_normals(W=1) row 0 normals: [[0.999201, -0.0, 0.039968], [0.999201, -0.0, 0.039968], [0.999201, -0.0, 0.039968], [0.999201, -0.0, 0.039968]]
depth_to_normals(H=1) shape: (1, 3, 1, 4)
depth_to_normals(H=W=1) shape: (1, 3, 1, 1)
```

## Expected

`.squeeze()` is replaced by an explicit axis removal (`squeeze(0)` / `squeeze(-2)` on the axes that are genuinely spurious, or a `reshape` to the intended `(*, H, W, 3)`), so `unproject_meshgrid(H, W, K)` returns `(B, H, W, 3)` for every `H` and `W` including 1. `depth_to_3d_v2` and `warp_frame_depth` then agree with `depth_to_3d` at `W = 1` as they already do everywhere else, and `depth_to_normals` returns the `(B, 3, H, 1)` normal map its documented layout asks for.

That is a focused fix and is welcome as a PR: the output is unchanged for every `W > 1`, and the regression test is `assert depth_to_3d_v2(d, K).shape == (1, 3, 1, 3)` for a `W = 1` depth, which fails today. `torch.equal(depth_to_3d(d, K)[0].permute(1, 2, 0)[None], depth_to_3d_v2(d[:, 0], K))` is the sharper pin — it holds at `H = 1` today and fails at `W = 1`.

This also removes the one known obstacle to aliasing `depth_to_3d` onto `depth_to_3d_v2` in the coordinated window: the duplication ledger's `depth_to_3d` / `depth_to_3d_v2` row records the `W = 1` divergence as the thing that must be fixed before the alias is possible.

## Classification

**Broken output** on a degenerate-but-valid shape — a wrong output shape with no error, and the `(*, H, W, 3)` contract violated. Idiom 4 (a degenerate input silently producing garbage instead of raising), and idiom 3 on the `KORNIA_CHECK_SHAPE` inside `depth_to_3d_v2` that passes on the corrupted grid.

## Disposition

n/a — single symbol. `depth_to_3d_v2` and `warp_frame_depth` are consequences of the one `unproject_meshgrid` line.

## Environment

kornia 0.9.0rc1 @ `92e5fe8d` (main), torch 2.14.0, python 3.11, cpu. Found during the camera conventions audit (batch 5, tracking issue #4262); the batch documents the behavior below as it is and pins it, and does not change it.

Posted on behalf of @ducha-aiki by Claude (Opus 5).

### PR DESCRIPTION
**Pull request by thegoodengineer**
## What does this pull request do?

`unproject_meshgrid` squeezed the grid it builds with a bare `.squeeze()`. `create_meshgrid` returns `(1, H, W, 2)`, so that call is meant to drop the leading batch axis, but it drops *every* size-1 axis. At `W == 1` the width axis goes with it, the grid broadcasts across a phantom width, and the documented `(*, H, W, 3)` contract breaks:

```text
unproject_meshgrid(H=1, W=3): (1, 1, 3, 3)
unproject_meshgrid(H=3, W=1): (1, 1, 3, 3)   same shape, different values
depth_to_3d(W=1)    shape: (1, 3, 3, 1)
depth_to_3d_v2(W=1) shape: (1, 3, 3, 3)      should be (1, 3, 1, 3)
warp_frame_depth(W=1) shape: (1, 2, 4, 4)    should be (1, 2, 4, 1)
```

`H = 1` is unaffected, because the batch axis absorbs the squeeze, and `H = W = 1` is correct by accident. `W = 1` is a degenerate but valid shape (a single image column, a 1-pixel-wide crop) and nothing raised.

The squeeze is now `squeeze(0)`, with a comment saying why the axis is named explicitly. Three regression tests are added, parametrized over degenerate and ordinary sizes: the grid shape itself, `depth_to_3d_v2` against `depth_to_3d` (the sharper pin, since the two agree everywhere else), and the `warp_frame_depth` output shape.

## Related issue or discussion

Closes #4278

## What did you check?

The new tests on unmodified `main` fail at `W = 1` and only there, which is exactly the reported shape:

```text
6 failed, 16 passed, 109 deselected
FAILED test_unproject_meshgrid_degenerate_sizes[cpu-float32-3-1] - assert torch.Size([2, 1, 3, 3]) == (2, 3, 1, 3)
FAILED test_depth_to_3d_v2_degenerate_sizes[cpu-float32-3-1]    - assert torch.Size([2, 3, 3, 3]) == (2, 3, 1, 3)
FAILED test_shape_degenerate_sizes[cpu-float32-4-1]             - assert torch.Size([1, 2, 4, 4]) == (1, 2, 4, 1)
```

Output for `W > 1` is unchanged. I compared `main` against this branch over a batched, asymmetric-`K`, float64 set of `unproject_meshgrid` (4x5, 7x2, 2x7, 13x11, both `normalize_points` values), `depth_to_3d_v2` and `warp_frame_depth` calls: `torch.equal` on all ten.

```text
$ pytest tests/geometry --dtype=float32 (excluding test_ransac.py)
====== 2593 passed, 50 skipped, 27 xfailed, 4 xpassed, 120 warnings in 85.46s ======

$ pytest tests/geometry/test_depth.py tests/geometry/test_depth_warper.py --dtype=float32,float64
================================ 157 passed in 2.34s ================================

$ ruff check / ruff format --diff on both changed files
All checks passed!
```

### CODE REVIEW

**Review by ducha-aiki — DISMISSED** [review]
Reviewed at `795628f4`: code approved.

The explicit squeeze(0) removes only the intended leading axis. All 22 focused cases pass. Against merge-base 3ea797d4, the W=1 grid, depth_to_3d_v2, depth_to_normals and warp cases reproduce the unintended expanded shapes; this head restores their width-one outputs. An ordinary W=5 asymmetric float64 grid/depth/warp comparison is byte-identical. No code findings. CI is green; please resolve the CHANGELOG-only conflict with current main before merging.

Validation ran in isolated checkouts with the repository Python environment (torch 2.14.0, CPU); CUDA was not tested locally.

Posted on behalf of [@ducha-aiki](https://github.com/ducha-aiki) by Codex (gpt-6-astra).

**Review by ducha-aiki — APPROVED** [review]
